test/Microsoft.AspNet.Mvc.FunctionalTests/TryValidateModelTest.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.TestHost;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.FunctionalTests
+{
+    public class TryValidateModelTest
+    {
+        private readonly IServiceProvider _services = TestHelper.CreateServices(nameof(FormatterWebSite));
+        private readonly Action<IApplicationBuilder> _app = new FormatterWebSite.Startup().Configure;
+
+        [Fact]
+        public async Task TryValidateModel_SimpleModelInvalidProperties()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+
+            // Act
+            var response = await client.GetAsync("http://localhost/TryValidateModel/GetInvalidUser");
+
+            // Assert
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+
+            var json = JsonConvert.DeserializeObject<Dictionary<string, string[]>>(
+                await response.Content.ReadAsStringAsync());
+            Assert.Equal("The field Id must be between 1 and 2000.",json["Id"][0]);
+            Assert.Equal(
+                "The field Name must be a string or array type with a minimum length of '5'.", json["Name"][0]);
```

Comment:
nit: these assertions won't catch cases where the JSON has additional properties (e.g. `json["something"]`) or errors (e.g. `json["Name"][1]`).  suggest comparing the returned value as a `string` to confirm entire response.